src/hooks/progress/use-update-tutorial-progress.ts

```diff
@@ -0,0 +1,150 @@
+import { useEffect, useState } from 'react'
+import { useInView } from 'react-intersection-observer'
+import useAuthentication from 'hooks/use-authentication'
+// Tutorial progress utilities
+import { progressLabelToPercent } from 'lib/learn-client/api/progress/formatting'
+import {
+	useTutorialProgress,
+	useTutorialProgressMutations,
+} from 'hooks/progress'
+// Types
+import {
+	TutorialIdCollectionId,
+	TutorialProgressLabel,
+} from 'lib/learn-client/types'
+import { TutorialProgressArgs } from './use-tutorial-progress-mutations/types'
+
+type IntersectionRef = (node?: Element) => void
+
+interface UpdateProgressReturnInterface {
+	startRef: IntersectionRef
+	endRef: IntersectionRef
+}
+
+/**
+ * Given a tutorialId and collectionId,
+ * Return two refs in an object, { startRef, endRef },
+ * which can be placed on elements to track tutorial progress.
+ *
+ * On initial load, regardless of startRef or endRef visibility,
+ * if the tutorial does not yet have recorded progress,
+ * the tutorial will be marked as `visited` (0 percent).
+ *
+ * When the startRef element moves out of the viewport,
+ * the tutorial will be marked as `in_progress` (50 percent).
+ *
+ * When the endRef element moves into the viewport,
+ * the tutorial will be marked as `complete` (100 percent).
+ */
+export function useUpdateTutorialProgress({
```

Comment:
This hook handles tracking progress, simplifying the consuming `TutorialView`'s responsibility to placing the `startRef` and `endRef` on elements. This is very similar to existing functionality in the `hashicorp/learn` repo.

Note: Acknowledging [early feedback in Slack to possibly combine this with `use-tutorial-progress-mutations`](https://hashicorp.slack.com/archives/C02T80JTNS2/p1661377009842339), I've ended up having this separate as I found combining them yielded a pretty huge file. Very open to other approaches though 🙇‍♂️